cmd/query/app/server.go

```diff
@@ -114,7 +129,26 @@ func createHTTPServer(querySvc *querysvc.QueryService, queryOpts *QueryOptions,
 	recoveryHandler := recoveryhandler.NewRecoveryHandler(logger, true)
 	return &http.Server{
 		Handler: recoveryHandler(handler),
+	}, nil
+}
+
+func (s *Server) serveHTTP(httpListener net.Listener) error {
+	var err error
+	if s.queryOptions.TLSHTTP.Enabled {
+		tlsCfg, err1 := s.queryOptions.TLSHTTP.Config()
+		if err1 == nil {
+			tlsCfg.Rand = rand.Reader
+			tlsHTTPListener := tls.NewListener(httpListener, tlsCfg)
```

Comment:
is this going to be a problem due to #2338 ?

Reply:
Here i am only testing the HTTP server component separately. i.e.  to check the `createHTTPServer`method and to check if TLSEnabled actually works .  The problem of #2338 occurs only when we want to test the queryServer which has both HTTPServer and GRPC server.